edgePy/DGEList.py

```diff
@@ -292,17 +292,40 @@ def rpkm(
         # self = self.cpm(log=log, prior_count=prior_count)
 
     def tpm(
-        self, transcripts: Mapping, log: bool = False, prior_count: float = PRIOR_COUNT
-    ) -> None:
-        """Return the DGEList normalized to reads per kilobase of transcript
-        length.
+        self,
+        gene_lengths: np.ndarray,
+        log: bool = False,
+        prior_count: float = PRIOR_COUNT,
+        mean_fragment_lengths: np.ndarray = None,
+    ) -> np.ndarray:
+        """Normalize the DGEList to transcripts per million
```

Comment:
Not quite. The way you have the function work now is that you are returning the normalized counts matrix (2D array) and not the normalized `DGEList` object.